src/Abstractions/NexusMods.Abstractions.Loadouts/Models/LoadoutItem.cs

```diff
@@ -35,4 +35,33 @@ public partial class LoadoutItem : IModelDefinition
     /// Optional parent of the item.
     /// </summary>
     public static readonly ReferenceAttribute<LoadoutItemGroup> Parent = new(Namespace, nameof(Parent)) { IsIndexed = true, IsOptional = true };
+
+    [PublicAPI]
+    public partial struct ReadOnly
+    {
+        /// <summary>
+        /// Returns true if this item or any of the item's parents are disables.
+        /// </summary>
+        public bool IsDisabledOrParentsDisabled()
+        {
+            var disabled = this.IsDisabled;
+            if (disabled)
+                return true;
+
+            var currentItem = this;
+            while (currentItem.HasParent())
+            {
+                currentItem = currentItem.Parent.AsLoadoutItem();
+                if (currentItem.IsDisabled)
+                    return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// True if this item contains a parent, else false.
+        /// </summary>
+        public bool HasParent() => this.Contains(LoadoutItem.Parent);
+    }
```

Comment:
Duplicate of https://github.com/Nexus-Mods/NexusMods.App/blob/abe8b6e9a43ef358ec2d6720be45746ba4e5a7a2/src/Abstractions/NexusMods.Abstractions.Loadouts/Extensions/LoadoutItemExtensions.cs#L58-L64.